### PR TITLE
[bugfix] Fix parsing logic when handling empty/missing fields

### PR DESCRIPTION
**Pull request by andyndang**
* Add tests with data from Java library

### CODE REVIEW

**Review by ibackus — COMMENTED** [review]
I'm not sure that this will work with dataset profiles serialized by the python code, unless there are previous changes I'm not aware of.  I believe we'll need to make those changes to all python writers.  I can test this.

If we are moving away from protobuf to delimited protobuf for serialization to disk, we should formalize that as our standard data format for all protobuf binary outputs.

I think there's probably value in centralizing this logic in utility functions for reading/writing delimited protobuf to avoid duplicate code for other protobuf classes.

**Comment by andyndang** [comment]
@ibackus the Python serialized data set is still deserializable with `from_protobuf_string` method (no change here).

We should definitely standarize the format to delimited format, but that's the next step. This delimited logic only applies to the top level message - inner messages don't use delimited deserialization

File: src/whylogs/util/dsketch.py [hunk 1]
```diff
@@ -211,7 +211,7 @@ def deserialize(x: bytes):
         If x is an empty sketch, None is returned
         """
         if len(x) <= 8:
-            FrequentItemsSketch()
+            return FrequentItemsSketch()
```
Comment (andyndang) [inline src/whylogs/util/dsketch.py:214]: @ibackus I think this was the bug in the existing code?

Comment (ibackus) [reply src/whylogs/util/dsketch.py:214]: That definitely seems to be a problem.

**Review by ibackus — DISMISSED** [review]

**Comment by andyndang** [comment]
@ibackus I added the delimited logic for DatasetProfiel (that's the only place we need delimited serialization)

**Pull request merged**